data/lib/service/tournament/tournament_service.dart

```diff
@@ -54,6 +54,61 @@ class TournamentService {
     }
   }
 
+  Stream<List<TournamentModel>> streamActiveTournaments() {
+    final currentDate = DateTime.now();
+    final date30DaysAgo = currentDate.subtract(Duration(days: 30));
+
+    return _tournamentCollection
+        .where(FireStoreConst.endDate, isGreaterThanOrEqualTo: date30DaysAgo)
+        .snapshots()
+        .asyncMap(
+      (event) async {
+        return await Future.wait(
+          event.docs.map(
+            (e) async {
+              final tournament = e.data();
+              final matchIds = tournament.match_ids;
+              if (matchIds.isNotEmpty) {
+                final matches = await _matchService.getMatchesByIds(matchIds);
+                final status = getTournamentStatus(matches);
+                return tournament.copyWith(matches: matches, status: status);
+              }
+              return tournament;
+            },
+          ),
+        );
+      },
+    ).handleError((error, stack) => throw AppError.fromError(error, stack));
+  }
+
+  Future<List<TournamentModel>> getTournaments({
```

Comment:
let's move it after createTournament function, we've following structure
- Create Function
- Get Functions
- Stream Functions
- Update Functions
- Delete Functions
- Helper Functions